engine/src/rgbmatrix.cpp

```diff
@@ -357,17 +362,21 @@ void RGBMatrix::setMapColors(RGBAlgorithm *algorithm)
     if (m_group == NULL)
         m_group = doc()->fixtureGroup(fixtureGroup());
 
-    if (m_group != NULL)
+    QVector<unsigned int> rawColors;
+    for (int i = 0; i < algorithm->acceptColors(); i++)
```

Comment:
```
const int acceptColors = algorithm->acceptColors();
rawColors.reserve(acceptColors);
for (int i = 0; i < acceptColors; i++)
```

this way only one vector allocation will happen, and algorithm->acceptColors() will be called just one. This is important because each RGBScript::acceptColors call locks mutex and calls javascript code